tests/glm/test_glm.py

```diff
@@ -1988,6 +1988,208 @@ def test_inputtype_std_errors(regression_data, categorical, split, fit_intercept
+
+    # Here, statsmodels does not do a degree of freedom adjustment,
+    # so we manually add it for nonrobust and robust errors
+    corr = len(y) / (len(y) - X.shape[1] - int(fit_intercept))
+
+    # nonrobust
+    # mdl.covariance_matrix_ = mdl_sm.fit(cov_type="nonrobust").cov_params()
+    our_results = mdl.wald_test_matrix(
+        R, r, X=X, y=y, dispersion=dispersion, robust=False
+    )
+    fit_sm = mdl_sm.fit(cov_type="nonrobust")
+    sm_results = fit_sm.wald_test(
+        (R, r), cov_p=fit_sm.cov_params() * corr, scalar=False
+    )
+
+    np.testing.assert_allclose(
+        our_results.test_statistic, sm_results.statistic, rtol=1e-3
+    )
+    np.testing.assert_allclose(our_results.p_value, sm_results.pvalue, atol=1e-3)
+    assert our_results.df == sm_results.df_denom
+
+    # robust
+    our_results = mdl.wald_test_matrix(R, r, X=X, y=y, robust=True)
+    fit_sm = mdl_sm.fit(cov_type="HC1")
+    sm_results = fit_sm.wald_test(
+        (R, r), cov_p=fit_sm.cov_params() * corr, scalar=False
+    )
+
+    np.testing.assert_allclose(
+        our_results.test_statistic, sm_results.statistic, rtol=1e-3
+    )
+    np.testing.assert_allclose(our_results.p_value, sm_results.pvalue, atol=1e-3)
+    assert our_results.df == sm_results.df_denom
+
+    # clustered
+    rng = np.random.default_rng(42)
+    clu = rng.integers(5, size=len(y))
+    our_results = mdl.wald_test_matrix(R, r, X=X, y=y, clusters=clu)
+    sm_fit = mdl_sm.fit(cov_type="cluster", cov_kwds={"groups": clu})
+    sm_results = sm_fit.wald_test((R, r), scalar=False)
+
+    np.testing.assert_allclose(
+        our_results.test_statistic, sm_results.statistic, rtol=1e-3
+    )
+    np.testing.assert_allclose(our_results.p_value, sm_results.pvalue, atol=1e-3)
+    assert our_results.df == sm_results.df_denom
+
+
+@pytest.mark.parametrize(
+    "R, r",
+    [
+        pytest.param(np.array([[0] * 9 + [1]]), np.array([0]), id="single"),
+        pytest.param(np.array([[0] * 8 + [1] * 2]), np.array([0]), id="multiple_vars"),
+        pytest.param(
+            np.array([[0] * 9 + [2], [0] * 8 + [1, 1]]),
+            np.array([0, 0]),
+            id="multiple_constraints",
+        ),
+        pytest.param(np.array([[0] * 9 + [1]]), np.array([2]), id="rhs_not_zero"),
+    ],
+)
+def test_wald_test_matrix_fixed_cov(regression_data, R, r):
+    X, y = regression_data
+
+    mdl = GeneralizedLinearRegressor(
+        alpha=0, family="gaussian", fit_intercept=False
+    ).fit(X=X, y=y, store_covariance_matrix=True)
+    mdl_sm = sm.GLM(endog=y, exog=X, family=sm.families.Gaussian())
+
+    # Use the same covariance matrix for both so that we can use tighter tolerances
+    our_results = mdl.wald_test_matrix(R, r)
+    fit_sm = mdl_sm.fit()
+    sm_results = fit_sm.wald_test((R, r), cov_p=mdl.covariance_matrix(), scalar=False)
+
+    np.testing.assert_allclose(
+        our_results.test_statistic, sm_results.statistic, rtol=1e-8
+    )
```

Comment:
Do we have any good rule of ascertaining how big the expected difference with statsmodels due to numerical imprecisions should be? I.e. are we sure that `1e-3` including recomputation of covariance matrix and `1e-8` without should be expected?

One way of checking this would be by comparing deviations between statsmodels and other packages (e.g., GLM in R) for the same problem and checking that they would be of the same order (no need for a test, could just be a one-off analysis).

Reply:
Good idea, I'll check against R